### PR TITLE
Feature: Add timeInterval_ms macro

### PR DESCRIPTION
**Pull request by karaolidis**
This will allow higher precision queries on DateTime64 columns.

### CODE REVIEW

**Review by asimpson — DISMISSED** [review]
Thanks for the contribution! :trophy:

File: pkg/macros/macros.go [hunk 1]
```diff
@@ -93,6 +93,15 @@ func TimeInterval(query *sqlds.Query, args []string) (string, error) {
 	return fmt.Sprintf("toStartOfInterval(toDateTime(%s), INTERVAL %d second)", args[0], int(seconds)), nil
 }
 
+func TimeIntervalMs(query *sqlds.Query, args []string) (string, error) {
+	if len(args) != 1 {
+		return "", fmt.Errorf("%w: expected 1 argument, received %d", sqlds.ErrorBadArgumentCount, len(args))
+	}
+
+	milliseconds := math.Max(float64(query.Interval.Milliseconds()), 1)
+	return fmt.Sprintf("toStartOfInterval(toDateTime(%s), INTERVAL %d millisecond)", args[0], int(milliseconds)), nil
```
Comment (mshustov) [inline pkg/macros/macros.go:102]: Maybe to [toDateTime64](https://clickhouse.com/docs/en/sql-reference/functions/type-conversion-functions#todatetime64)  if you need sub-second precision? What scale would you set, then?

Comment (karaolidis) [reply pkg/macros/macros.go:102]: Indeed, I missed that. If we want milliseconds to keep it consistent with other macros then I assume `toDateTime64(3, %s)`?

**Review by asimpson — APPROVED** [review]
Great call out @mshustov,  this looks good to me :trophy:

**Review by mshustov — APPROVED** [review]

File: pkg/macros/macros.go (hunk 1 above)
Comment (mshustov) [reply pkg/macros/macros.go]: sounds good

**Comment by asimpson** [comment]
Thanks again @karaolidis :trophy:

**Comment by karaolidis** [comment]
@mshustov @asimpson I just tested out the PR and it turns out that the syntax for `toDateTime64` is `toDateTime64(%s, 3)`, not `toDateTime64(3, %s)`. Should I make a new PR with the fix or are hotfixes on `main` allowed? Sorry for not testing it before the merge, I got confused while reading the `DateTime64` [docs](https://clickhouse.com/docs/en/sql-reference/data-types/datetime64) :facepalm:

```diff
diff --git a/pkg/macros/macros.go b/pkg/macros/macros.go
index de73dd7..caad257 100644
--- a/pkg/macros/macros.go
+++ b/pkg/macros/macros.go
@@ -99,7 +99,7 @@ func TimeIntervalMs(query *sqlds.Query, args []string) (string, error) {
 	}
 
 	milliseconds := math.Max(float64(query.Interval.Milliseconds()), 1)
-	return fmt.Sprintf("toStartOfInterval(toDateTime64(3, %s), INTERVAL %d millisecond)", args[0], int(milliseconds)), nil
+	return fmt.Sprintf("toStartOfInterval(toDateTime64(%s, 3), INTERVAL %d millisecond)", args[0], int(milliseconds)), nil
 }
 
 func IntervalSeconds(query *sqlds.Query, args []string) (string, error) {
diff --git a/pkg/macros/macros_test.go b/pkg/macros/macros_test.go
index 1b3d9a5..f8b0eec 100644
--- a/pkg/macros/macros_test.go
+++ b/pkg/macros/macros_test.go
@@ -122,7 +122,7 @@ func TestMacroTimeIntervalMs(t *testing.T) {
 	}
 	got, err := macros.TimeIntervalMs(&query, []string{"col"})
 	assert.Nil(t, err)
-	assert.Equal(t, "toStartOfInterval(toDateTime64(3, col), INTERVAL 20000 millisecond)", got)
+	assert.Equal(t, "toStartOfInterval(toDateTime64(col, 3), INTERVAL 20000 millisecond)", got)
 }
 
 func TestMacroIntervalSeconds(t *testing.T) {
```

**Comment by mshustov** [comment]
@karaolidis yes, please, open a PR with a hotfix

**Pull request merged**